mlcommons_box/mlcommons_box/common/objects/base.py

```diff
@@ -0,0 +1,197 @@
+import abc
+import collections
+import logging
+import os
+import typing
+
+import mlspeclib
+
+logger = logging.getLogger(__name__)
+
+
+class BaseField(abc.ABC):
```

Comment:
what is abc.ABC ?

Reply:
this is the "abstract base class" that allows us to define abstract classes/methods